cpp/src/arrow/compute/exec/ir_consumer.cc

```diff
@@ -152,14 +152,14 @@ struct ConvertLiteralImpl {
     switch (t.id()) {
       case Type::DECIMAL128: {
         std::array<uint64_t, 2> little_endian;
-        std::memcpy(little_endian.data(), lit->value(), lit->value()->size());
+        std::memcpy(little_endian.data(), lit->value()->data(), lit->value()->size());
```

Comment:
This still seems incorrect. `little_endian` is default-initialized but, as the `std::array` docs [note](https://en.cppreference.com/w/cpp/container/array), 
> default initialization may result in indeterminate values for non-class T

... meaning that unless `lit->value->size()` is large enough, some decimal data may still be incorrect.

@bkietz Am I wrong?

Reply:
The size of the literal string is asserted to be equal to the decimal's byte_width above, so every byte of `little_endian` should be safely overwritten

Reply:
Woops, I missed that, sorry.